lib/question.js

```diff
@@ -646,7 +647,201 @@ module.exports = {
             });
         });
     },
+    // Might want to do some refactoring since most of the code is the same from _gradeVariantWithClient()
+    _regradeVariantWithClient(client, variant, submission, question, course, authn_user_id, callback) {
```

Comment:
Actually, I agree with you that we shouldn't have this function at all, instead we should modify `_gradeVariantWithClient()` so that it takes an optional `submission` argument and skips the `variants_select_submission_for_grading` if it already has a submission. That would let us unify `_gradeVariantWithClient()` and `_regradeVariantWithClient()`.